### PR TITLE
mirror: add baremetal-installer image mapping

### PR DESCRIPTION
**Pull request by stbenjam**
This adds the mapping for baremetal-installer.

### CODE REVIEW

File: core-services/image-mirroring/openshift/mapping_origin_4_2 [hunk 1]
```diff
@@ -78,6 +78,7 @@ registry.svc.ci.openshift.org/origin/4.2:hypershift quay.io/openshift/origin-hyp
 registry.svc.ci.openshift.org/origin/4.2:insights-operator quay.io/openshift/origin-insights-operator:4.2 quay.io/openshift/origin-support-operator:4.2.0 quay.io/openshift/origin-insights-operator:latest
 registry.svc.ci.openshift.org/origin/4.2:installer quay.io/openshift/origin-installer:4.2 quay.io/openshift/origin-installer:4.2.0 quay.io/openshift/origin-installer:latest
 registry.svc.ci.openshift.org/origin/4.2:installer-artifacts quay.io/openshift/origin-installer-artifacts:4.2 quay.io/openshift/origin-installer-artifacts:4.2.0 quay.io/openshift/origin-installer-artifacts:latest
+registry.svc.ci.openshift.org/origin/4.2:baremetal-installer quay.io/openshift/origin-baremetal-installer:4.2 quay.io/openshift/origin-baremetal-installer:4.2.0 quay.io/openshift/origin-baremetal-installer:latest
```
Comment (smarterclayton) [inline core-services/image-mirroring/openshift/mapping_origin_4_2]: The file is sorted alphabetically, please put this in the correct location.

Comment (stbenjam) [reply core-services/image-mirroring/openshift/mapping_origin_4_2]: There are others out of place. Shall I fix them?

Comment (stbenjam) [reply core-services/image-mirroring/openshift/mapping_origin_4_2]: I updated this to put baremetal-installer in the right place, but local-storage-static-provisioner and sriov-network-config-daemon are not in the right place. I can push a follow-up PR if you want

**Comment by stbenjam** [comment]
@smarterclayton Fixed

**Pull request merged**